db/system_keyspace.cc

```diff
@@ -267,6 +267,7 @@ schema_ptr system_keyspace::topology() {
             .with_column("enabled_features", set_type_impl::get_instance(utf8_type, true), column_kind::static_column)
             .with_column("session", uuid_type, column_kind::static_column)
             .with_column("tablet_balancing_enabled", boolean_type, column_kind::static_column)
+            .with_column("tablets_per_shard_goal", int32_type, column_kind::static_column)
             .with_column("upgrade_state", utf8_type, column_kind::static_column)
```

Comment:
Do we need to do anything for upgrades?

Reply:
We're good unless someone tries to override the default via RESTful API in a mixed cluster, in which case the update may be lost. I could add a cluster feature to fail the update in a mixed cluster.

Reply:
If the upgrade is incomplete it's okay to behave as if it hadn't happened.